include/caffe/layer.hpp

```diff
@@ -184,6 +184,25 @@ class Layer {
   }
 
   /**
+   * @brief Sets blobs
+   */
+  void SetBlobs(const vector<Blob<Dtype>*>& weights) {
```

Comment:
For both of these, Layer::blobs() is sufficient, there's no change needed to this file.